src/test/php/PHPMD/AbstractTestCase.php

```diff
@@ -406,12 +406,11 @@ protected function getMethodMock($metric = null, $value = null)
     /**
      * Creates a mocked function node instance.
      *
-     * @param string $metric The metric acronym used by PHP_Depend.
+     * @param ?string $metric The metric acronym used by PHP_Depend.
      * @param ?numeric $value The expected metric return value.
-     * @return FunctionNode&MockObject
      * @throws Throwable
      */
-    protected function createFunctionMock($metric = null, $value = null)
+    protected function createFunctionMock(?string $metric = null, mixed $value = null): FunctionNode&MockObject
```

Comment:
Actually `?numeric` could be narrowed to:
```suggestion
    protected function createFunctionMock(?string $metric = null, string|int|float|null $value = null): FunctionNode&MockObject
```

Reply:
Yeah, I prefer not to though as it still isn't as narrow as the PHPDoc type and feel clunkey with all the unions.